### PR TITLE
Feat:AutoLogin

### PR DESCRIPTION
**Pull request by Bibitha-V**

### CODE REVIEW

**Review by theproductiveprogrammer — CHANGES_REQUESTED** [review]

File: login.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            return console.log(err);
+          }
+        })
+}
+
+function getUserLoginInfo(){
+    var login_json=null
+    try {
+        if(fs.existsSync(path.join(loc.home(),"logininfo.json"))) {
+            login_json=  fs.readFileSync(path.join(loc.home(),"logininfo.json"),"utf-8", (err, data) => {
```
[the remark this answers is not in the document]
Comment (Bibitha-V) [reply login.js:25]: updated with readFile method

File: web/login.js [hunk 2]
```diff
@@ -129,6 +126,34 @@ function dappURL(u) {
...
 }
 
+ function auto(log,store,usr,pwd){
+    setTimeout(function(){
+      let serverURL = store.get("settings.serverURL")
+      let u = dappURL(serverURL) + "/login" 
+      req.post(u, {usr,pwd}, (err, resp) => {
+      if(err) {
+        log("err/login", err)
+        alert("auto Login failed")
```
[the remark this answers is not in the document]
Comment (Bibitha-V) [reply web/login.js:136]: updated

File: web/login.js [hunk 3]
```diff
@@ -129,6 +126,34 @@ function dappURL(u) {
...
+      }
+      let ui = resp.body
+      if(invalid_1(ui)) {
+        log("err/login/resp/invalid", resp)
+        alert("auto Login failed")
```
[the remark this answers is not in the document]
Comment (Bibitha-V) [reply web/login.js:141]: here also updated the same

File: web/main.js [hunk 4]
```diff
@@ -69,10 +68,22 @@ function showUI(log, store) {
...
-    if(!ui) curr.page = login.e(log, curr.store)
-    else curr.page = home.e(ui, log, curr.store)
-
-    main.c(curr.page)
+    if(!ui) {
+      window.autologin.getLoginInfo().then(function(result) {
+      if(result == null){       
+          curr.page =  login.e(log, curr.store)
+          main.c(curr.page)
+          document.body.firstElementChild.remove()
```
[the remark this answers is not in the document]
Comment (Bibitha-V) [reply web/main.js:76]: removed by using id

File: web/main.js [hunk 5]
```diff
@@ -69,10 +68,22 @@ function showUI(log, store) {
...
+        }
+        else{
+          result=JSON.parse(result)
```
[the remark this answers is not in the document]
Comment (Bibitha-V) [reply web/main.js:79]: handled in getLoginInfo

File: index.js [hunk 6]
```diff
@@ -131,6 +134,20 @@ function setupIPC(log) {
...
+    login.saveLoginInfo(u,p)
+  })
+
+  ipcMain.handle("get-logininfo", async () => {
+    return await new Promise((resolve, reject) => {
+      login.getUserLoginInfo((data) => {
+        resolve(data);
+      });
+    }).then((result) => {
+      return JSON.parse(result);
```
Comment (theproductiveprogrammer) [inline index.js:148]: JSON.parse should always be wrapped in a try-catch block because it can throw errors

Comment (theproductiveprogrammer) [reply index.js:148]: Also the JSON.parse is better done within getUserLoginInfo (login info should be the actual info not a string that we need to parse after)

Comment (Bibitha-V) [reply index.js]: updated the code

**Review by theproductiveprogrammer — APPROVED** [review]

**Pull request merged**